### ISSUE TITLE
Generated md5sum does not match expected md5sum for sdl2 recipe

### ISSUE THREAD

**Issue by davied9** [kivy/python-for-android#1318]
I'm working on build kivy app for android with buildozer/p4a in [buildozer vm](https://kivy.org/#download)

buildozer : 0.34
p4a : 0.6.0

build with error

![image](https://user-images.githubusercontent.com/29981898/43676215-4d3b1b06-981f-11e8-8f2a-d838dbdc61f0.png)

found following code in python-for-android\pythonforandroid\recipes\sdl2\__init__.py
```
    version = "2.0.4"
    url = "https://www.libsdl.org/release/SDL2-{version}.tar.gz"
    md5sum = '44fc4a023349933e7f5d7a582f7b886e'
```

so I download the file and check , oops ...
![image](https://user-images.githubusercontent.com/29981898/43676249-0a145ad0-9820-11e8-86f9-94cfe8ee1433.png)

how could I fix this ? (I've tried old_toolchain, there is something else bothering me, so I checked back)

my source code is very simple
```
from kivy.app import App
from kivy.uix.button import Button


class SimpleApp(App):

    def build(self):
        return Button(text='joke')

if '__main__' == __name__:
    SimpleApp().run()
```

and my build command is 
```
buildozer --verbose android debug
```

**Comment by davied9** [comment]
After fix this, I continuing meet similar problems for 

python-2.7.2.tar.bz2
```
  RAN: /bin/tar xf /home/kivy/Desktop/Projects/kivy_cracker/EG002_BuildSimpleApp/.buildozer/android/platform/build/packages/hostpython2/python-2.7.2.tar.bz2

  STDOUT:


  STDERR:
/bin/tar: This does not look like a tar archive

gzip: stdin: not in gzip format
/bin/tar: Child returned status 1
/bin/tar: Error is not recoverable: exiting now
```

 SDL2_image-2.0.1.tar.gz
```
  RAN: /bin/tar xf /home/kivy/Desktop/Projects/kivy_cracker/EG002_BuildSimpleApp/.buildozer/android/platform/build/packages/sdl2_image/SDL2_image-2.0.1.tar.gz

  STDOUT:


  STDERR:
/bin/tar: This does not look like a tar archive

gzip: stdin: not in gzip format
/bin/tar: Child returned status 1
/bin/tar: Error is not recoverable: exiting now
```

I've fixed python-2.7.2 for the same way ( i.e. wget ), but no lucky for sdl2_image


Here is more defailed log for SDL2_image if I don't download it with wget :
```
[INFO]:    Downloading pyjnius
[DEBUG]:   -> running mkdir -p /home/kivy/Desktop/Projects/kivy_cracker/EG002_BuildSimpleApp/.buildozer/android/platform/build/packages/pyjnius
[INFO]:    -> directory context /home/kivy/Desktop/Projects/kivy_cracker/EG002_BuildSimpleApp/.buildozer/android/platform/build/packages/pyjnius
[DEBUG]:   -> running basename https://github.com/kivy/pyjnius/archive/master.zip
[DEBUG]:   	master.zip
[INFO]:    pyjnius download already cached, skipping
[INFO]:    <- directory context /home/kivy/Desktop/OpenSource/kivy/python-for-android
[INFO]:    Downloading kivy
[DEBUG]:   -> running mkdir -p /home/kivy/Desktop/Projects/kivy_cracker/EG002_BuildSimpleApp/.buildozer/android/platform/build/packages/kivy
[INFO]:    -> directory context /home/kivy/Desktop/Projects/kivy_cracker/EG002_BuildSimpleApp/.buildozer/android/platform/build/packages/kivy
[DEBUG]:   -> running basename https://github.com/kivy/kivy/archive/1.10.0.zip
[DEBUG]:   	1.10.0.zip
[INFO]:    kivy download already cached, skipping
[INFO]:    <- directory context /home/kivy/Desktop/OpenSource/kivy/python-for-android
[INFO]:    # Building all recipes for arch armeabi-v7a
[INFO]:    # Unpacking recipes
[INFO]:    Unpacking hostpython2 for armeabi-v7a
[DEBUG]:   -> running basename https://python.org/ftp/python/2.7.2/Python-2.7.2.tar.bz2
[DEBUG]:   	Python-2.7.2.tar.bz2
[INFO]:    -> directory context /home/kivy/Desktop/Projects/kivy_cracker/EG002_BuildSimpleApp/.buildozer/android/platform/build/build/other_builds/hostpython2/desktop
[INFO]:    hostpython2 is already unpacked, skipping
[INFO]:    <- directory context /home/kivy/Desktop/OpenSource/kivy/python-for-android
[INFO]:    Unpacking sdl2_image for armeabi-v7a
[DEBUG]:   -> running basename https://www.libsdl.org/projects/SDL_image/release/SDL2_image-2.0.1.tar.gz
[DEBUG]:   	SDL2_image-2.0.1.tar.gz
[INFO]:    -> directory context /home/kivy/Desktop/Projects/kivy_cracker/EG002_BuildSimpleApp/.buildozer/android/platform/build/build/bootstrap_builds/sdl2_gradle-python2/jni
Traceback (most recent call last):
  File "/usr/lib/python2.7/runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/home/kivy/Desktop/OpenSource/kivy/python-for-android/pythonforandroid/toolchain.py", line 978, in <module>
    main()
  File "/home/kivy/Desktop/OpenSource/kivy/python-for-android/pythonforandroid/toolchain.py", line 975, in main
    ToolchainCL()
  File "/home/kivy/Desktop/OpenSource/kivy/python-for-android/pythonforandroid/toolchain.py", line 512, in __init__
    getattr(self, args.subparser_name.replace('-', '_'))(args)
  File "/home/kivy/Desktop/OpenSource/kivy/python-for-android/pythonforandroid/toolchain.py", line 149, in wrapper_func
    build_dist_from_args(ctx, dist, args)
  File "/home/kivy/Desktop/OpenSource/kivy/python-for-android/pythonforandroid/toolchain.py", line 193, in build_dist_from_args
    build_recipes(build_order, python_modules, ctx)
  File "pythonforandroid/build.py", line 559, in build_recipes
  File "pythonforandroid/recipe.py", line 520, in prepare_build_dir
  File "pythonforandroid/recipe.py", line 422, in unpack
  File "/usr/local/lib/python2.7/dist-packages/sh.py", line 1245, in __call__
    return RunningCommand(cmd, call_args, stdin, stdout, stderr)
  File "/usr/local/lib/python2.7/dist-packages/sh.py", line 633, in __init__
    self.wait()
  File "/usr/local/lib/python2.7/dist-packages/sh.py", line 651, in wait
    self.handle_command_exit_code(exit_code)
  File "/usr/local/lib/python2.7/dist-packages/sh.py", line 672, in handle_command_exit_code
    raise exc
sh.ErrorReturnCode_2: 

  RAN: /bin/tar xf /home/kivy/Desktop/Projects/kivy_cracker/EG002_BuildSimpleApp/.buildozer/android/platform/build/packages/sdl2_image/SDL2_image-2.0.1.tar.gz

  STDOUT:


  STDERR:
/bin/tar: This does not look like a tar archive

gzip: stdin: not in gzip format
/bin/tar: Child returned status 1
/bin/tar: Error is not recoverable: exiting now

# Command failed: /usr/bin/python -m pythonforandroid.toolchain create --dist_name=myapp --bootstrap=sdl2 --requirements=kivy --arch armeabi-v7a --copy-libs --color=always --storage-dir=/home/kivy/Desktop/Projects/kivy_cracker/EG002_BuildSimpleApp/.buildozer/android/platform/build
# 
# Buildozer failed to execute the last command
# The error might be hidden in the log above this error
# Please read the full log, and search for it before
# raising an issue with buildozer itself.
# In case of a bug report, please add a full log with log_level = 2
```

here is log if I download sdl2_image mannually with wget from  https://www.libsdl.org/projects/SDL_image/release/SDL2_image-2.0.1.tar.gz
```
[INFO]:    Downloading kivy
[DEBUG]:   -> running mkdir -p /home/kivy/Desktop/Projects/kivy_cracker/EG002_BuildSimpleApp/.buildozer/android/platform/build/packages/kivy
[INFO]:    -> directory context /home/kivy/Desktop/Projects/kivy_cracker/EG002_BuildSimpleApp/.buildozer/android/platform/build/packages/kivy
[DEBUG]:   -> running basename https://github.com/kivy/kivy/archive/1.10.0.zip
[DEBUG]:   	1.10.0.zip
[INFO]:    kivy download already cached, skipping
[INFO]:    <- directory context /home/kivy/Desktop/OpenSource/kivy/python-for-android
[INFO]:    # Building all recipes for arch armeabi-v7a
[INFO]:    # Unpacking recipes
[INFO]:    Unpacking hostpython2 for armeabi-v7a
[DEBUG]:   -> running basename https://python.org/ftp/python/2.7.2/Python-2.7.2.tar.bz2
[DEBUG]:   	Python-2.7.2.tar.bz2
[INFO]:    -> directory context /home/kivy/Desktop/Projects/kivy_cracker/EG002_BuildSimpleApp/.buildozer/android/platform/build/build/other_builds/hostpython2/desktop
[INFO]:    hostpython2 is already unpacked, skipping
[INFO]:    <- directory context /home/kivy/Desktop/OpenSource/kivy/python-for-android
[INFO]:    Unpacking sdl2_image for armeabi-v7a
[DEBUG]:   -> running basename https://www.libsdl.org/projects/SDL_image/release/SDL2_image-2.0.1.tar.gz
[DEBUG]:   	SDL2_image-2.0.1.tar.gz
[INFO]:    -> directory context /home/kivy/Desktop/Projects/kivy_cracker/EG002_BuildSimpleApp/.buildozer/android/platform/build/build/bootstrap_builds/sdl2_gradle-python2/jni
Traceback (most recent call last):
  File "/usr/lib/python2.7/runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/home/kivy/Desktop/OpenSource/kivy/python-for-android/pythonforandroid/toolchain.py", line 978, in <module>
    main()
  File "/home/kivy/Desktop/OpenSource/kivy/python-for-android/pythonforandroid/toolchain.py", line 975, in main
    ToolchainCL()
  File "/home/kivy/Desktop/OpenSource/kivy/python-for-android/pythonforandroid/toolchain.py", line 512, in __init__
    getattr(self, args.subparser_name.replace('-', '_'))(args)
  File "/home/kivy/Desktop/OpenSource/kivy/python-for-android/pythonforandroid/toolchain.py", line 149, in wrapper_func
    build_dist_from_args(ctx, dist, args)
  File "/home/kivy/Desktop/OpenSource/kivy/python-for-android/pythonforandroid/toolchain.py", line 193, in build_dist_from_args
    build_recipes(build_order, python_modules, ctx)
  File "pythonforandroid/build.py", line 559, in build_recipes
  File "pythonforandroid/recipe.py", line 520, in prepare_build_dir
  File "pythonforandroid/recipe.py", line 422, in unpack
  File "/usr/local/lib/python2.7/dist-packages/sh.py", line 1245, in __call__
    return RunningCommand(cmd, call_args, stdin, stdout, stderr)
  File "/usr/local/lib/python2.7/dist-packages/sh.py", line 633, in __init__
    self.wait()
  File "/usr/local/lib/python2.7/dist-packages/sh.py", line 651, in wait
    self.handle_command_exit_code(exit_code)
  File "/usr/local/lib/python2.7/dist-packages/sh.py", line 672, in handle_command_exit_code
    raise exc
sh.ErrorReturnCode_2: <exception str() failed>
# Command failed: /usr/bin/python -m pythonforandroid.toolchain create --dist_name=myapp --bootstrap=sdl2 --requirements=kivy --arch armeabi-v7a --copy-libs --color=always --storage-dir=/home/kivy/Desktop/Projects/kivy_cracker/EG002_BuildSimpleApp/.buildozer/android/platform/build
# 
# Buildozer failed to execute the last command
# The error might be hidden in the log above this error
# Please read the full log, and search for it before
# raising an issue with buildozer itself.
# In case of a bug report, please add a full log with log_level = 2
```

If there is anything to provide more info, just let me know, thanks

**Comment by inclement** [comment]
I haven't seen anything about this recently, it seems to have been fixed.
